### PR TITLE
Makes the chemical smoke machine use stock parts for things

### LINKED ISSUE #33453: Smoke machine's radius cannot be increased beyond 9 when upgraded.
It contains the following parts:

 Smoke Machine (Machine Board).

 Some glass.

 A quadratic capacitor.

 A bluespace matter bin.

 A bluespace matter bin.

 A femto-manipulator.

 A bluespace power cell.

Max radius was 9. The machine was wrenched down. Clicking any number above 9 did not change the radius.

### PR DESCRIPTION
**Pull request by nicbn**
[Changelogs]: #
:cl: nicbn
fix: Now the chem smoke machine uses stock parts for volume and range.
/:cl:

Closes #33499 
Fixes #33453

### CODE REVIEW

**Review by Cyberboss — DISMISSED** [review]

**Review by Cyberboss — CHANGES_REQUESTED** [review]
The UI needs an update to reflect this still

**Comment by Robustin** [comment]
No no no.

It was already nerfed by 70% of range and volume with the concept that we could bring some of that range and volume BACK with parts.

Now you're gutting an already overnerfed toy AGAIN just to make the fucking parts even more mandatory. That's was, and is, the absolute peak of RND cancer where you had to gobble science's balls just to get your machines into a decent condition. 

Also the title is misleading, it already uses stock parts and it affects the efficiency of chemical usage. 

**Make it scale ABOVE the current levels with better parts, otherwise fuck right off with that "nerf machine to make it dependent on an upgrade" bullshit.**

**Comment by davethwave** [comment]
So far from my pitiful insight of code this PR nerfs the smoke machine so it can only have 100 instead of 300 chems and also nerfs the range but I can't tell by how much.

**Review by Cyberboss — CHANGES_REQUESTED** [review]
Also remove the nerfs

**Comment by nicbn** [comment]
Ok

**Comment by nicbn** [comment]
Uh where do I find tgui documentation

**Comment by Buggy123** [comment]
Thank you for doing this but don't nerf it more, just let stock parts improve it.

**Comment by vuonojenmustaturska** [comment]
@nicbn there is none, but i've worked with it extensively so #coderbus

**Comment by Buggy123** [comment]
I can't support this unless the baseline nerf is removed. Leave the machine's stats at their _original_ values (volume 300, area 9), and simply allow better parts to _increase_ them. The smoke machine is in a acceptable state as-is and does not need a nerf to justify the ability to upgrade it.

I'm not the only person who's levied this criticism either, please change it. Infact literally everyone who has expressed any opinion at all about the nerfs has requested you remove them, including a _maintainer_.

**Comment by nicbn** [comment]
I'm going to remove the nerf calm down men

**Comment by Buggy123** [comment]
its literally 2 digits why would it take so long

**Comment by nicbn** [comment]
Calm down bro
It's not like the game is unplayable without this fix

**Review by Cyberboss — DISMISSED** [review]

**Comment by nicbn** [comment]
Hey I didn't do the tgui thing yet
That shouldn't be approved

**Comment by nicbn** [comment]
My PR is ready

**Comment by nicbn** [comment]
@Cyberboss

File: code/modules/reagents/chemistry/machinery/smoke_machine.dm [hunk 1]
```diff
@@ -28,10 +27,11 @@
 	opaque = FALSE
 	alpha = 100
 
-
 /obj/machinery/smoke_machine/Initialize()
 	. = ..()
-	create_reagents(volume)
+	create_reagents(100)
+	for(var/obj/item/stock_parts/matter_bin/B in component_parts)
+		reagents.maximum_volume += 100 * B.rating
```
Comment (ninjanomnom) [inline code/modules/reagents/chemistry/machinery/smoke_machine.dm]: keep the volume var and when you calculate based on parts do `initial(volume) * B.rating` so in the future only the one place has to be edited to buff or nerf

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

File: code/modules/reagents/chemistry/machinery/smoke_machine.dm (hunk 1 above)
Comment (nicbn) [reply code/modules/reagents/chemistry/machinery/smoke_machine.dm]: I'll try another approach, see if it is ok.

**Comment by ninjanomnom** [comment]
If you're not going to change the var then a define is more appropriate.

**Review by ninjanomnom — APPROVED** [review]

**Comment by Robustin** [comment]
Ugh how did this get merged I barely understand code but I could tell it was a shitshow, what do the setting buttons even DO now?

**Comment by nicbn** [comment]
They set the range

**Comment by nicbn** [comment]
I don't understand what is a shitshow here

**Pull request merged**